akka-http-core/src/main/scala/akka/http/javadsl/settings/ParserSettings.scala

```diff
@@ -56,7 +56,7 @@ abstract class ParserSettings private[akka] () extends BodyPartParser.Settings {
   def withMaxHeaderNameLength(newValue: Int): ParserSettings = self.copy(maxHeaderNameLength = newValue)
   def withMaxHeaderValueLength(newValue: Int): ParserSettings = self.copy(maxHeaderValueLength = newValue)
   def withMaxHeaderCount(newValue: Int): ParserSettings = self.copy(maxHeaderCount = newValue)
-  def withMaxContentLength(newValue: Long): ParserSettings = self.copy(maxContentLength = newValue)
+  def withMaxContentLength(newValue: Long): ParserSettings = self.copy(maxContentLengthSetting = Some(newValue))
```

Comment:
```suggestion
  def withMaxContentLength(newValue: Long): ParserSettings = self.copy(maxContentLengthSetting = Option(newValue))
```

Reply:
Isn't that just doing an extra unnecessary null check?

Reply:
Yes, my thinking is that this is the javadsl and it doesn't contain a variant to unset the value.

Reply:
Ah, I see. That still won't work as `scala.Long` is the primitive type. But we can introduce an explicitly method for unsetting.